libs/libdaemon/src/BaseDaemon.cpp

```diff
@@ -788,10 +788,12 @@ void BaseDaemon::closeFDs()
 #endif
     if (proc_path.isDirectory()) /// Hooray, proc exists
     {
-        Poco::DirectoryIterator itr(proc_path), end;
-        for (; itr != end; ++itr)
+        std::vector<std::string> fds;
+        /// in /proc/self/fd directory filenames are numeric file descriptors
+        proc_path.list(fds);
```

Comment:
lgtm as a temporary fix as this routine looks dodgy anyway :)

Reply:
Same code can be found in python sources, for example https://github.com/python/cpython/blob/6f9bc72c79c3262e5d0f2c0e96b016477399cfb1/Modules/_posixsubprocess.c#L211-L383.